federation/pkg/federation-controller/dns/dns.go

```diff
@@ -175,179 +303,109 @@ func getResolvedEndpoints(endpoints []string) ([]string, error) {
 	return resolvedEndpoints, nil
 }
 
-/* ensureDnsRrsets ensures (idempotently, and with minimum mutations) that all of the DNS resource record sets for dnsName are consistent with endpoints.
-   if endpoints is nil or empty, a CNAME record to uplevelCname is ensured.
-*/
-func (s *ServiceController) ensureDnsRrsets(dnsZone dnsprovider.Zone, dnsName string, endpoints []string, uplevelCname string) error {
+// ensureDNSRrsets ensures (idempotently, and with minimum mutations) that all of the DNS resource record sets for
+// dnsName are consistent with endpoints. if endpoints is nil or empty, a CNAME record to uplevelCname is ensured.
+func (dc *DNSController) ensureDNSRrsets(dnsZone dnsprovider.Zone, service *apiv1.Service, addRecords bool) (
```

Comment:
The original of this was more loosely coupled (only dnsZone and strings as parameters - even though it was a method of ServiceController, that was never used and should have been deleted).  What is your reasoning behind changing to a more tightly coupled implementation? (i.e. service as a parameter)?  It's not clear what the benefits are of all these code changes and additional coupling are?